pkg/stanza/fileconsumer/internal/reader/reader.go

```diff
@@ -164,7 +165,7 @@ func (r *Reader) Close() *Metadata {
 func (r *Reader) Read(dst []byte) (int, error) {
 	// Skip if fingerprint is already built
 	// or if fingerprint is behind Offset
-	if len(r.Fingerprint.FirstBytes) == r.FingerprintSize || int(r.Offset) > len(r.Fingerprint.FirstBytes) {
+	if r.Fingerprint.IsMaxSize(r.FingerprintSize, r.Offset) {
```

Comment:
I think I mentioned this before, but offset should not be a concern of fingerprints. Is there really any benefit to this change?